src/ast.ts

```diff
@@ -31,6 +31,11 @@ export function asNode(ast: AST | undefined): Node | undefined {
     }
 }
 
+// Some nodes are "virtual" (generated by the serialization processs) and do not
+// have proper properties. We set the parent properties to these, which also aid
+// the language server.
+const nodesThatNeedParentProperties: string[] = ['ID', 'ExpField'];
```

Comment:
Is this just an optimization? Since the `motoko` npm package is used for other use cases besides the VS Code extension, it would be best if we could find a solution that's independent of the language server implementation details.

Reply:
I wouldn't call it an intentional optimization, although it probably is better than doing it doing serialization.

I wouldn't call it an implementation detail of the language server, either; anyone using these nodes would find that these nodes do not have properties without this, and it's useful to add them. Although I do agree that other "virtual" nodes are still missing them.

So, probably it would be best to do it in `motoko` during serialization, because there it's easier to identify which nodes are missing them, even though performance-wise it would probably not be as good as doing it here.